libs/flags/json_flag.go

```diff
@@ -33,7 +35,27 @@ func (j *JsonFlag) Unmarshal(v any) error {
 	if j.raw == nil {
 		return nil
 	}
-	return json.Unmarshal(j.raw, v)
+
+	dv, err := jsonloader.LoadJSON(j.raw)
+	if err != nil {
+		return err
+	}
+
+	err = convert.ToTyped(v, dv)
+	if err != nil {
+		return err
+	}
+
+	_, diags := convert.Normalize(v, dv)
+	if len(diags) > 0 {
+		summary := ""
+		for _, diag := range diags {
+			summary += fmt.Sprintf("- %s\n", diag.Summary)
+		}
+		return fmt.Errorf("json input error:\n%v", summary)
+	}
```

Comment:
Is there significance to normalizing before/after the `ToTyped` call?

The conversion logic now lives in `ToTyped`.

Reply:
@pietern ToTyped seems not to return diagnostics for, for example, unknown fields while normalise do, hence the reason for extra call